src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -159,60 +172,80 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
+            {
+                // delegateParameter may be null, consider this case: Action x = (a, b) => { };
+                // we will still fall back to object
+
+                if (parameterNode.Type == null)
+                {
+                    parameterNode = parameterNode.WithType(delegateParameter?.Type.GenerateTypeSyntax() ?? s_objectType);
+                }
+
+                if (delegateParameter?.HasExplicitDefaultValue == true)
+                {
+                    parameterNode = parameterNode.WithDefault(GetDefaultValue(delegateParameter));
+                }
+
+                return parameterNode;
+            }
+        }
+
+        private static ParameterListSyntax TryGetOrCreateParameterList(AnonymousFunctionExpressionSyntax anonymousFunction)
         {
             switch (anonymousFunction)
             {
                 case SimpleLambdaExpressionSyntax simpleLambda:
-                    return GenerateSimpleLambdaParameterList(semanticModel, simpleLambda, delegateType.DelegateInvokeMethod, cancellationToken);
+                    return SyntaxFactory.ParameterList(SyntaxFactory.SingletonSeparatedList(simpleLambda.Parameter));
                 case ParenthesizedLambdaExpressionSyntax parenthesizedLambda:
-                    return GenerateParenthesizedLambdaParameterList(semanticModel, parenthesizedLambda, delegateType.DelegateInvokeMethod, cancellationToken);
+                    return parenthesizedLambda.ParameterList;
+                case AnonymousMethodExpressionSyntax anonymousMethod:
+                    return anonymousMethod.ParameterList; // may be null!
                 default:
                     throw ExceptionUtilities.UnexpectedValue(anonymousFunction);
             }
         }
 
-        private ParameterListSyntax GenerateSimpleLambdaParameterList(
-            SemanticModel semanticModel, SimpleLambdaExpressionSyntax lambdaExpression, IMethodSymbol delegateInvokeMethod, CancellationToken cancellationToken)
+        private static InvocationExpressionSyntax WithNewParameterNames(InvocationExpressionSyntax invocation, IMethodSymbol method, ParameterListSyntax newParameterList)
         {
-            var parameter = semanticModel.GetDeclaredSymbol(lambdaExpression.Parameter, cancellationToken);
-            var type = parameter?.Type.GenerateTypeSyntax() ?? s_objectType;
-
-            var parameterSyntax = SyntaxFactory.Parameter(lambdaExpression.Parameter.Identifier).WithType(type);
-            var param = delegateInvokeMethod.Parameters[0];
-            if (param.HasExplicitDefaultValue)
+            return invocation.ReplaceNodes(invocation.ArgumentList.Arguments, (argumentNode, _) =>
             {
-                parameterSyntax = parameterSyntax.WithDefault(GetDefaultValue(param));
-            }
+                if (argumentNode.NameColon == null || argumentNode.NameColon.IsMissing)
```

Comment:
can't have a missing NameColon.  It should either be there or not.

Reply:
then why is the check here? http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Extensions/ArgumentSyntaxExtensions.cs,73

Reply:
No clue.  That code is *old* :)   It's likely it existed at a point in time with the syntax model where such a thing was possible.  Feel free to fix this up as well.